metric-anomaly-data-model/src/main/java/org/hypertrace/alert/engine/metric/anomaly/datamodel/queue/KafkaQueueAlertTaskConsumer.java

```diff
@@ -0,0 +1,77 @@
+package org.hypertrace.alert.engine.metric.anomaly.datamodel.queue;
+
+import com.typesafe.config.Config;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Properties;
+import jdk.jshell.spi.ExecutionControl.NotImplementedException;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.hypertrace.alert.engine.metric.anomaly.datamodel.AlertTask;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KafkaQueueAlertTaskConsumer implements Queue<AlertTask> {
```

Comment:
should the task consumer also be generic?

class KafkaQueueTaskConsumer<T> implements Queue<T>